examples/tensorflow/mnist.yaml

```diff
@@ -0,0 +1,22 @@
+apiVersion: jobset.x-k8s.io/v1alpha2
+kind: JobSet
+metadata:
+  name: tensorflow
+spec:
+  replicatedJobs:
+  - name: tensorflow
+    template:
+      spec:
+        parallelism: 2
+        completions: 2
+        backoffLimit: 5
+        template:
+          spec:
+            containers:
+            - name: tensorflow
+              image: docker.io/kubeflowkatib/tf-mnist-with-summaries:latest
```

Comment:
just noticed this is using kubeflow, which imo kind of defeats the purpose of using jobset, since kubeflow is presumably creating [TFJobs](https://www.kubeflow.org/docs/components/training/tftraining/#what-is-tfjob) under the hood, a CRD which encapsulates the TF distributed training primitives like the leader, workers, parameter server, which when modeled as a JobSet would each be a different ReplicatedJob, so they can have different success/failure policies applied to them and so on. 

It's also handling the setting of the [TF_CONFIG](https://www.tensorflow.org/guide/distributed_training#setting_up_the_tf_config_environment_variable) env var which is something the user still needs to configure when using JobSets, although automating the configuration of these kinds of env vars is part of the future plans for JobSet.

Reply:
So the image is mostly just a KubeFlow hosted image that builds/deploys tensorflow. It’s not using KubeFlow.  

If you’d prefer to just host a tensorflow image in Jobset (like we did for PyTorch) we could go that route.  I have very little experience with tensorflow so I was grabbing workable examples from the internet.

Reply:
Yeah, if the user wants to use this image I don't see any reason to use JobSet here, as none of its features are being used, it's just running a single Indexed Job. Is the headless service being utilized via pod-to-pod communication via pod hostnames at least? What happens when we set `enableDNSHostnames: false` to explicitly not create the headless service, does the example still work?

Reply:
I'll have to check.  Where is your source for the pytorch image btw?

Reply:
I don't have the Dockerfile or code for it published publicly but I can share it somewhere. Perhaps I should include the training script and Dockerfile in examples/pytorch along with the JobSet yaml that is already there?

Reply:
Yea I think that is good idea IMO.  I notice that these images/code can code rot pretty quickly.  

I brought the tensorflow example being out-of-date on kubeflow ie https://github.com/kubeflow/training-operator/issues/1884.

Looks like they have the source defined in the repo and some pushes to the images.  Maybe we don't need to go that far but it would be nice to have a way to up-date these if they rot too much.

Reply:
I think this is fine, the image has nothing to do with kubeflow itself, it is just the training script.